test/js/composed-parent-node-test.js

```diff
@@ -0,0 +1,182 @@
+// Copyright 2015 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function() {
+    module('composedParentNode');
+
+    test('No Shadow DOM', function() {
+        var fixture = document.getElementById('qunit-fixture');
+        var parent = fixture.appendChild(document.createElement('div'));
+        parent.id = 'parent';
+        var child = parent.appendChild(document.createElement('span'));
+        child.id = 'child';
+        equal(axs.dom.composedParentNode(child), parent);
+        var textNode = child.appendChild(document.createTextNode('Hello, world!'));
+        equal(axs.dom.composedParentNode(textNode), child);
+    });
+
+    test('Shadow root parent', function() {
+        var fixture = document.getElementById('qunit-fixture');
+        var host = fixture.appendChild(document.createElement('div'));
+        host.id = 'host';
+        if (host.createShadowRoot) {
+            var root = host.createShadowRoot();
+            equal(axs.dom.composedParentNode(root), host);
+        } else {
+            console.warn('Test platform does not support shadow DOM');
+            ok(true);
+        }
+    });
+
+    test('Distributed content parent', function() {
+        var fixture = document.getElementById('qunit-fixture');
+        var host = fixture.appendChild(document.createElement('div'));
+        host.id = 'host';
+        var testElement = host.appendChild(document.createElement('span'));
+        if (host.createShadowRoot) {
+            var root = host.createShadowRoot();
+            var content = document.createElement('content');
+            root.appendChild(content);
+            equal(axs.dom.composedParentNode(testElement), host);
+        } else {
+            console.warn('Test platform does not support shadow DOM');
+            ok(true);
+        }
+    });
+
+    test('Content not direct child of Shadow root', function() {
```

Comment:
If you'd like to test re-projection, you can duplicate this test and add a shadowRoot+<content> to `shadowParent`.